### PR TITLE
Add docments to General API and Debugging sections of 06_data.block n…

### PR DESCRIPTION
**Pull request by butchland**
Add docments for the following sections
* General API
   - DataBlock
   - datasets
   - dataloaders
 * Debugging
   - summary

### CODE REVIEW

**Comment by butchland** [comment]
@warner-benjamin @muellerzr any updates on when you plan review or merge?

File: fastai/data/block.py [hunk 1]
```diff
@@ -107,25 +115,41 @@ def __init__(self, blocks=None, dl_type=None, getters=None, n_inp=None, item_tfm
...
         pv(f"{len(splits)} datasets of sizes {','.join([str(len(s)) for s in splits])}", verbose)
         return Datasets(items, tfms=self._combine_type_tfms(), splits=splits, dl_type=self.dl_type, n_inp=self.n_inp, verbose=verbose)
 
-    def dataloaders(self, source, path='.', verbose=False, **kwargs):
+    def dataloaders(self,
+        source,  # The data source
+        path:str='.', # The path to the data source
+        verbose:bool=False, # Show verbose messages
+        **kwargs
+    ):
```
Comment (warner-benjamin) [inline fastai/data/block.py:152]: `path` also has a sneaky feature where it sets the `Learner` path location too
```suggestion
    def dataloaders(self,
        source,  # The data source
        path:str='.', # Data source and default `Learner` path
        verbose:bool=False, # Show verbose messages
        **kwargs
    ):
```

**Review by warner-benjamin — CHANGES_REQUESTED** [review]
Left a few suggestions and corrections. If you could replicate the suggestions the rest of the doc strings, everything else looks good.

File: fastai/data/block.py [hunk 2]
```diff
@@ -170,7 +194,12 @@ def _find_fail_collate(s):
 
 # Cell
 @patch
-def summary(self: DataBlock, source, bs=4, show_batch=False, **kwargs):
+def summary(self: DataBlock,
```
Comment (kevinbird15) [inline fastai/data/block.py:197]:
```suggestion
def summary(self:DataBlock,
```

File: fastai/data/block.py [hunk 3]
```diff
@@ -107,25 +115,41 @@ def __init__(self, blocks=None, dl_type=None, getters=None, n_inp=None, item_tfm
...
+    ):
         self.source = source                     ; pv(f"Collecting items from {source}", verbose)
         items = (self.get_items or noop)(source) ; pv(f"Found {len(items)} items", verbose)
         splits = (self.splitter or RandomSplitter())(items)
         pv(f"{len(splits)} datasets of sizes {','.join([str(len(s)) for s in splits])}", verbose)
         return Datasets(items, tfms=self._combine_type_tfms(), splits=splits, dl_type=self.dl_type, n_inp=self.n_inp, verbose=verbose)
 
-    def dataloaders(self, source, path='.', verbose=False, **kwargs):
+    def dataloaders(self,
+        source,  # The data source
```
Comment (kevinbird15) [inline fastai/data/block.py:148]:
```suggestion
        source, # The data source
```

File: fastai/data/block.py [hunk 4]
```diff
@@ -107,25 +115,41 @@ def __init__(self, blocks=None, dl_type=None, getters=None, n_inp=None, item_tfm
     def _combine_type_tfms(self): return L([self.getters, self.type_tfms]).map_zip(
         lambda g,tt: (g.fs if isinstance(g, Pipeline) else L(g)) + tt)
 
-    def new(self, item_tfms=None, batch_tfms=None):
+    def new(self,
+        item_tfms:list=None,  # List of `ItemTransform`s, applied on an item
```
Comment (kevinbird15) [inline fastai/data/block.py:119]:
```suggestion
        item_tfms:list=None, # List of `ItemTransform`s, applied on an item
```

File: fastai/data/block.py [hunk 5]
```diff
@@ -107,25 +115,41 @@ def __init__(self, blocks=None, dl_type=None, getters=None, n_inp=None, item_tfm
...
+    ):
         if getters is None: getters = L(ItemGetter(i) for i in range(2 if blocks is None else len(L(blocks))))
         get_items = _zip if get_items is None else compose(get_items, _zip)
         return cls(blocks=blocks, getters=getters, get_items=get_items, **kwargs)
 
-    def datasets(self, source, verbose=False):
+    def datasets(self,
+        source, # The data source
+        verbose:bool=False, # Show verbose messages
+    ):
```
Comment (kevinbird15) [inline fastai/data/block.py:140]:
```suggestion
    ) -> Datasets:
```

File: fastai/data/block.py (hunk 1 above)
Comment (kevinbird15) [inline fastai/data/block.py:152]:
```suggestion
    ) -> DataLoaders:
```

**Review by kevinbird15 — CHANGES_REQUESTED** [review]
nothing too big, just a few formatting nits and a few return ideas

File: fastai/data/block.py [hunk 6]
```diff
@@ -77,7 +77,15 @@ class DataBlock():
     blocks,dl_type = (TransformBlock,TransformBlock),TfmdDL
     _methods = 'get_items splitter get_y get_x'.split()
     _msg = "If you wanted to compose several transforms in your getter don't forget to wrap them in a `Pipeline`."
-    def __init__(self, blocks=None, dl_type=None, getters=None, n_inp=None, item_tfms=None, batch_tfms=None, **kwargs):
+    def __init__(self,
+        blocks:list=None, # List of `TransformBlock`s
+        dl_type:TfmdDL=None, # Specific class that inherits `TfmdDL`
```
Comment (kevinbird15) [inline fastai/data/block.py:82]:
```suggestion
        dl_type:TfmdDL=None, # Task-specific `TfmdDL`, defaults to `block`'s dl_type or `TfmdDL` if none specified
```
Not sure I like my version, but tried to convey the hierarchy of how dl_type is populated: `__init__.dl_type` > `block.dl_type` > `self.dl_type`

File: fastai/data/block.py [hunk 7]
```diff
@@ -13,10 +13,10 @@
...
     "A basic wrapper that links defaults transforms for the data block API"
     def __init__(self,
-        type_tfms:list=None, # List of `Transform`s
-        item_tfms:list=None, # List of `ItemTransform`s, applied on an item
-        batch_tfms:list=None, # List of `Transform`s or `RandTransform`s, applied by batch
-        dl_type:TfmdDL=None, # Specific class that inherits `TfmdDL`
+        type_tfms:list=None, # One or more `Transform`s
+        item_tfms:list=None, # `ItemTransform`s, applied on an item
+        batch_tfms:list=None, # `Transform`s or `RandTransform`s, applied by batch
+        dl_type:TfmdDL=None, # Task specific `TfmdDL`, defaults to `TfmdDL` if none specified
```
Comment (warner-benjamin) [inline fastai/data/block.py:19]:
```suggestion
        type_tfms:list=None, # One or more `Transform`s
        item_tfms:list=None, # `ItemTransform`s, applied on an item
        batch_tfms:list=None, # `Transform`s or `RandTransform`s, applied by batch
        dl_type:TfmdDL=None, # Task specific `TfmdDL`, defaults to `TfmdDL`
```

File: fastai/data/block.py [hunk 8]
```diff
@@ -78,12 +78,12 @@ class DataBlock():
     _methods = 'get_items splitter get_y get_x'.split()
     _msg = "If you wanted to compose several transforms in your getter don't forget to wrap them in a `Pipeline`."
     def __init__(self,
-        blocks:list=None, # List of `TransformBlock`s
-        dl_type:TfmdDL=None, # Specific class that inherits `TfmdDL`
-        getters:list=None, # List of getter functions
+        blocks:list=None, # One or more `TransformBlock`s
+        dl_type:TfmdDL=None, # Task specific `TfmdDL`, defaults to `block`'s dl_type or`TfmdDL` if none specified
+        getters:list=None, # Getter functions applied to results of `get_items`
```
Comment (warner-benjamin) [inline fastai/data/block.py:83]:
```suggestion
        blocks:list=None, # One or more `TransformBlock`s
        dl_type:TfmdDL=None, # Task specific `TfmdDL`, defaults to `block`'s dl_type or`TfmdDL`
        getters:list=None, # Getter functions applied to results of `get_items`
```

File: nbs/06_data.block.ipynb [hunk 9]
```diff
@@ -228,12 +228,12 @@
     "    _methods = 'get_items splitter get_y get_x'.split()\n",
     "    _msg = \"If you wanted to compose several transforms in your getter don't forget to wrap them in a `Pipeline`.\"\n",
     "    def __init__(self, \n",
-    "        blocks:list=None, # List of `TransformBlock`s\n",
-    "        dl_type:TfmdDL=None, # Specific class that inherits `TfmdDL` \n",
-    "        getters:list=None, # List of getter functions\n",
+    "        blocks:list=None, # One or more `TransformBlock`s\n",
+    "        dl_type:TfmdDL=None, # Task specific `TfmdDL`, defaults to `block`'s dl_type or`TfmdDL` if none specified\n",
+    "        getters:list=None, # Getter functions applied to results of `get_items`\n",
```
Comment (warner-benjamin) [inline nbs/06_data.block.ipynb:233]:
```suggestion
    "        blocks:list=None, # One or more `TransformBlock`s\n",
    "        dl_type:TfmdDL=None, # Task specific `TfmdDL`, defaults to `block`'s dl_type or`TfmdDL`\n",
    "        getters:list=None, # Getter functions applied to results of `get_items`\n",
```

File: nbs/06_data.block.ipynb [hunk 10]
```diff
@@ -70,10 +70,10 @@
...
     "    \"A basic wrapper that links defaults transforms for the data block API\"\n",
     "    def __init__(self, \n",
-    "        type_tfms:list=None, # List of `Transform`s\n",
-    "        item_tfms:list=None, # List of `ItemTransform`s, applied on an item\n",
-    "        batch_tfms:list=None, # List of `Transform`s or `RandTransform`s, applied by batch\n",
-    "        dl_type:TfmdDL=None, # Specific class that inherits `TfmdDL` \n",
+    "        type_tfms:list=None, # One or more `Transform`s\n",
+    "        item_tfms:list=None, # `ItemTransform`s, applied on an item\n",
+    "        batch_tfms:list=None, # `Transform`s or `RandTransform`s, applied by batch\n",
+    "        dl_type:TfmdDL=None, # Task specific `TfmdDL`, defaults to `TfmdDL` if none specified \n",
```
Comment (warner-benjamin) [inline nbs/06_data.block.ipynb:76]:
```suggestion
    "        type_tfms:list=None, # One or more `Transform`s\n",
    "        item_tfms:list=None, # `ItemTransform`s, applied on an item\n",
    "        batch_tfms:list=None, # `Transform`s or `RandTransform`s, applied by batch\n",
    "        dl_type:TfmdDL=None, # Task specific `TfmdDL`, defaults to `TfmdDL`\n",
```

**Review by warner-benjamin — APPROVED** [review]
I think we can remove the if none specified, as its implied by the defaults. Otherwise looks good to me.

**Review by kevinbird15 — APPROVED** [review]
Agreed with Benjamin's last if none specified point then I think this is ready to merge!

**Pull request merged**